staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -943,6 +974,36 @@ func (g *genValidations) emitValidationForChild(c *generator.Context, thisChild
 				case types.Struct:
 					// Call the type's validation function.
 					g.emitCallToOtherTypeFunc(c, fld.node, bufsw)
+					for _, subchild := range fld.inner {
+						if len(subchild.name) == 0 {
+							panic(fmt.Sprintf("missing child name for field in %v", thisNode))
+						}
+						if len(subchild.jsonName) == 0 {
+							panic(fmt.Sprintf("missing child JSON name for field %v.%s", thisNode, subchild.name))
+						}
+
+						leafType, typePfx, exprPfx := getLeafTypeAndPrefixes(subchild.childType)
+						targs := targs.WithArgs(generator.Args{
+							"inType":       fld.childType,
+							"fieldName":    subchild.name,
+							"fieldJSON":    subchild.jsonName,
+							"fieldType":    leafType,
+							"fieldTypePfx": typePfx,
+							"fieldExprPfx": exprPfx,
+						})
+						bufsw.Do("// field $.inType|raw$.$.fieldName$\n", targs)
+						bufsw.Do("errs = append(errs,\n", targs)
+						bufsw.Do("  func(obj, oldObj $.fieldTypePfx$$.fieldType|raw$, fldPath *$.field.Path|raw$) (errs $.field.ErrorList|raw$) {\n", targs)
+
+						if !subchild.fieldValidations.Empty() {
+							emitCallsToValidators(c, subchild.fieldValidations.Functions, bufsw)
+						}
+
+						bufsw.Do("    return\n", targs)
+						bufsw.Do("  }($.fieldExprPfx$obj.$.fieldName$, $.fieldExprPfx$oldObj.$.fieldName$, fldPath.Child(\"$.fieldJSON$\"))...)\n", targs)
```

Comment:
oldObj needs to be guarded with safe.Field or risk a nil deref.  You can see it in the generated code:  `&oldObj.StringField` - oldObj is nil for CREATE.

See `emitValidationForChild()`

Reply:
Done, now have safe.Field in the generated `inner` output:
```
			// field T2.StringField
			errs = append(errs,
				func(obj, oldObj *string, fldPath *field.Path) (errs field.ErrorList) {
					errs = append(errs, validate.FixedResult(opCtx, fldPath, obj, oldObj, false, "inner T1.T2.StringField")...)
					return
				}(&obj.StringField, safe.Field(oldObj, func(oldObj *T2) *string { return &oldObj.StringField }), fldPath.Child("stringField"))...) // < --- SAFE.FIELD THERE NOW
```